lib/chef/provider/directory.rb

```diff
@@ -64,7 +64,13 @@ def define_resource_requirements
               is_parent_writable = lambda do |base_dir|
                 base_dir = ::File.dirname(base_dir)
                 if ::File.exists?(base_dir)
-                  Chef::FileAccessControl.writable?(base_dir)
+                  if Chef::FileAccessControl.writable?(base_dir)
+                    true
+                  elsif Gem::Version.new(node['platform_version']) >= Gem::Version.new('10.11') && Chef::Util::PathHelper.is_sip_path?(base_dir)
```

Comment:
Putting a darwin check into path_helper is pretty trivial, but the node object isn't available to path_helper as far as In can tell.

I don't see a ruby constant that shows the kernel info for the system, otherwise I could look for the kernel version and map that to 10.11+.

I'll keep looking, but if you know of a good place I could grab this info that is available within chef-config, I'll use that.